_drafts/circle-ci-chibios.md

```diff
@@ -0,0 +1,438 @@
+Larger firmware projects sometimes have tens, even hundreds of engineers working in a single codebase. This introduces new challenges: builds break, lesser used features get forgotten about, new features miss support for older hardware platforms. Without solid tooling and processes, this can slow development to a grind.
+
+In the software world, the established solution is Continuous Integration. A hallmark of engineering at large tech companies like Facebook and Google, continuous integration has found its way into firmware projects, such as [Equisense](https://medium.com/equisense/firmware-quality-assurance-continuous-delivery-125884194ea5).
+
+<!-- excerpt start -->
+
+In this post, we introduce Continuous Integration as a modern technique to allow larger teams to move fast without breaking things. We’ll explain what Continuous Integration is, why you may want to use it, and walk you through setting up CircleCI on a firmware project.
+
+<!-- excerpt end -->
+
+This is the first post in our Building Better Firmware series. Future posts will cover testing techniques, test driven development, fuzzing, and continuous deployment.
+
+{:.no_toc}
+
+## Table of Contents
+
+<!-- prettier-ignore -->
+auto-gen TOC:
+{:toc}
+
+## What is Continuous Integration?
+Strictly speaking, Continuous Integration is the practice of regularly integrating code into a mainline repository. In other words: merging your work into master as often as possible.
+
+To achieve Continuous Integration, teams rely on a well established set of techniques:
+
+1. Revision Control: Continuous Integration depends on a solid Version Control System. Nowadays, it often is a git repository, but mercurial (hg), perforce, or SVN are fine options.
+2. Automated Builds: 
+3. Automated Tests
+
+These techniques have become synonymous with Continuous Integration, as they are universally leveraged as the way to achieve it.
+
+## Why Continuous Integration
+As complexity grows:
+1. Finding when and where an issue was introduced gets more difficult
+2. Side effects are less well understood, making it more likely that 
+3. Testing every feature and hardware configuration takes more and more time. 
+
+_Like Interrupt? [Subscribe](http://eepurl.com/gpRedv) to get our latest posts
+straight to your mailbox_
+
+## Continuous Integration Systems
+Many tools exist today that teams use to implement Continuous Integration. You may have heard of Jenkins, a popular open source tool, but many other exists: TravisCI, CircleCI, TeamCity, Bamboo, Build Bot, …
+
+Fundamentally, continuous integration systems all do roughly the same thing:
+1. Monitor your repository for changesets (either pull requests, or merges)
+2. For each changeset, execute a set of build & test steps
+3. Report the success or failure of build & test steps
+4. Save the resulting artifacts (e.g. firmware binary) for future use
+
+A good continuous integration system for firmware development should have the following features:
```

Comment:
nit: You have 4 numbered lists in a row. This one seems like a good candidate for a bullet list.